src/runtime/divert.rs

```diff
@@ -97,36 +106,24 @@ enum DivertType {
 
 impl From<DivertData> for Divert {
     fn from(divert: DivertData) -> Self {
-        match divert.divert_type {
-            DivertType::Standard { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new()
-            },
-            DivertType::Variable { target, var: () } => Self {
-                target: Some(TargetType::VarName(target)),
-                is_conditional: divert.conditional,
-                ..Self::new()
-            },
-            DivertType::Function { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new_function()
-            },
-            DivertType::Tunnel { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new_tunnel()
-            },
+        let new_divert = match divert.divert_type {
+            DivertType::Standard { path } => Self::new_divert(TargetType::Path(path)),
```

Comment:
How would you feel about removing the constructors for each divert type? I don't see them used anywhere else but here and imo it would be somewhat cleaner to have Divert::new set the `stack_push_type` based on the given TargetType (we could then maybe implement a `From<DivertType> for TargetType`?) I'm not sure how that would work with `external_args` for external function though

Reply:
Re: what to do with `external_args`

Would it make sense for them to be moved into `TargetType::ExternalName`'s data as a tuple with the name? It looks like the external args are only ever used with the external_function anyways so it makes sense to me

Reply:
I've inlined all the functions in a new commit, except for `new()` itself which I think is useful. And I think the `external_args` suggestion is a good idea, I've committed basically that.